react/ProductList.tsx

```diff
@@ -20,9 +20,11 @@ export function getProductList(products?: Product[]) {
   }
 
   const baseUrl = getBaseUrl()
-
+  const slash = baseUrl && baseUrl?.slice(-1) === '/' ? '' : '/'
   const productItems: ListItem[] = products.map((product, index) => {
-    const productUrl = `${baseUrl}/${product.link ?? `${product.linkText}/p`}`
+    let link = product.link ?? `${product.linkText}/p`
```

Comment:
In the provided workspace I'm seeing URLs like this in the LD+JSON: https://teststructuredata--itwhirlpool.myvtex.com/https://portal.vtexcommercestable.com.br/piano-cottura-a-induzione-in-vetroceramica-whirlpool-wl-b2977-ne-869991580870/p

This is because `product.link` is a full URL and it's being prepended with the `baseUrl`. I suggest removing it:

```suggestion
    let link = `${product.linkText}/p`
```

This may also solve the problem of the duplicated slashes, which I was unable to reproduce.